staging -> production: DIST alerts with forest filters and encoded tiles titiler support

solomon-negusse opened the pull request:
<!-- Please refer to our contributing documentation for any questions on submitting a pull request, or let us know here if you need any help -->

## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [ ] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [ ] Check the commit's or even all commits' message styles matches our requested structure.
- [ ] Check your code additions will fail neither code linting checks nor unit test.



## Pull request type

<!-- Please do not submit updates to dependencies unless it fixes an issue. --> 

<!-- Please try to limit your pull request to one type, submit multiple pull requests if needed. --> 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?
<!-- Please describe the behavior or changes that are being added by this PR. -->

-
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

<!-- If this introduces a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

<!-- Any other information that is important to this PR such as screenshots of how the component looks before and after the change. -->

solomon-negusse committed:
refactor integrated alerts algo and add dist alerts

solomon-negusse committed:
use namedtuple both for immutability and access with labels

solomon-negusse committed:
refactor to modularize and slim down __call__ method

solomon-negusse committed:
encapsulate alpha logic in a method

solomon-negusse committed:
add unit tests for alerts titiler algorithm

solomon-negusse committed:
serve encoded rgb to support Flagship date slider

solomon-negusse committed:
reuse alert config named tuple definitions; rename deforestationalert->alert to account for disturbance alerts

solomon-negusse committed:
move s3 creds/bucket name to env variable

solomon-negusse committed:
fix start date

solomon-negusse committed:
fix typo

solomon-negusse committed:
fix typo

solomon-negusse committed:
Merge pull request #164 from wri/dist-alert-algo

refactor integrated alerts algo and add dist alerts

solomon-negusse committed:
merge

solomon-negusse committed:
Fix typo

Co-authored-by: Gary Tempus <gtempus@users.noreply.github.com>

solomon-negusse committed:
add documentation and log message

solomon-negusse committed:
Merge branch 'titiler/encoded-alert-tiles' of github.com:wri/gfw-tile-cache into titiler/encoded-alert-tiles

solomon-negusse committed:
Merge pull request #166 from wri/titiler/encoded-alert-tiles

Titiler/encoded alert tiles

solomon-negusse committed:
Merge branch staging into dev

solomon-negusse committed:
Merge pull request #167 from wri/dev

Dev

solomon-negusse committed:
Merge branch 'production' into staging

dmannarino reviewed (dismissed):
Sweet refactor! Thanks for all the documentation, tests!

solomon-negusse committed:
dont use tilefactory for algos for ease of customization; make titiler endpoints consistent with tile cache service

solomon-negusse committed:
make encoded the default render_type

solomon-negusse committed:
don't use default alertconfidence enum; create separate alertconf enum for dist/integrated alerts as we surface in api doc

solomon-negusse committed:
update titiler test to point to new endpoint

solomon-negusse committed:
remove unused implementation query param from titiler endpoint

solomon-negusse committed:
add comment in the order of bands in tile

solomon-negusse committed:
add forest filter layers

solomon-negusse committed:
add DIST alerts forest mask test

solomon-negusse committed:
add tcl mask that was left behind

solomon-negusse committed:
consistent route file naming

solomon-negusse committed:
use the version enum

solomon-negusse committed:
use consistent import alias

solomon-negusse committed:
use color/conf and rendertype defaults

solomon-negusse committed:
fix import

solomon-negusse committed:
test test asset

solomon-negusse committed:
update latest versions in test

solomon-negusse committed:
fix list order

solomon-negusse committed:
Merge pull request #169 from wri/gtc-3020/update-titile-endpoints

make titiler tile endpoints consistent with tile cache service

solomon-negusse committed:
Merge branch 'dev' into gtc-3025/filter-alerts-by-forest

solomon-negusse committed:
clean up tree cover loss logic and comment

solomon-negusse committed:
simplify tree cover loss doc

solomon-negusse committed:
fix TCL mask for DIST alert

solomon-negusse committed:
update docs for clarity; add todo comment for version update in prod

solomon-negusse committed:
get DIST alerts filter dataset/versions

solomon-negusse committed:
Merge pull request #170 from wri/gtc-3025/filter-alerts-by-forest

Gtc 3025/filter alerts by forest

solomon-negusse committed:
Merge branch 'staging' into dev

solomon-negusse committed:
Merge pull request #171 from wri/dev

Dev -> staging:  DIST alerts forest filter

dmannarino committed:
Bump dependencies, pin psycopg2 which I couldn't get to build

dmannarino committed:
Merge pull request #172 from wri/bump_deps

Bump dependencies for security and starlette changes

danscales reviewed (dismissed):
Seems fine.   Though I'm not totally sure if we need to push the dist alerts changes to production yet.  Did you want to wait a bit and see if we have any more bug fixes or adjustments to endpoints, etc.?  Whatever you think is best is fine.

solomon-negusse committed:
update dataset name; allow test datasets

solomon-negusse committed:
add titiler endpoint to cloudfront

solomon-negusse committed:
add clarifying comment for the two titiler paths

solomon-negusse committed:
Merge pull request #173 from wri/update-dist-alerts-dataset

update dataset name; allow test datasets

solomon-negusse committed:
fix tree cover loss year field type; update var name to make mypy happy

solomon-negusse committed:
Merge pull request #174 from wri/fix-field-type

fix tree cover loss year field type; update var name to make mypy happy

solomon-negusse committed:
consistent and descriptive forest filter query params; detailed decoding doc

solomon-negusse committed:
doc styling; add alert decoding example

solomon-negusse committed:
doc updates

solomon-negusse committed:
Merge pull request #175 from wri/field-name-updates

consistent and descriptive forest filter query params; detailed decod…

solomon-negusse committed:
minor doc fix

gtempus committed:
:bug: fix: implementation replacement on redirect (#176)

* :bug: fix: Only replace implementation with `dynamic`

Supports FLAG-1126

* :white_check_mark: test: add initial pytest suite

* :art: refactor: create more declarative methods and tests

* :memo: docs: add docstring to handler

* :memo: docs: add implementation note regarding relative URLs

gtempus committed:
:bug: fix: implementation replacement on redirect (#176) (#178)

* :bug: fix: Only replace implementation with `dynamic`

Supports FLAG-1126

* :white_check_mark: test: add initial pytest suite

* :art: refactor: create more declarative methods and tests

* :memo: docs: add docstring to handler

* :memo: docs: add implementation note regarding relative URLs

(cherry picked from commit 8d79ee75271bd3b57879144722e4fa0f0cfcecef)

solomon-negusse committed:
swith from lambda to titiler tiling for integrated alerts

solomon-negusse committed:
update titiler endpoint in text

jterry64 committed:
Merge pull request #179 from wri/gtc-3033/integrated-alerts-titiler

switch from lambda to titiler tiling for integrated alerts

jterry64 committed:
Merge branch 'staging' into dev

jterry64 committed:
Merge pull request #182 from wri/dev

Dev to staging

danscales committed:
Fix "Tile index out of bounds" error for very north tiles

We are hitting the edges of the whole-world bounding box for very north
tiles, and need to add some rounding to deal with slight imprecision in
the floating point computations.

danscales committed:
Merge pull request #184 from wri/fix_validate_bbox

Fix "Tile index out of bounds" error for very north tiles

danscales commented:
Not needed, I merged the dist alert changes separately.
